script/generate_and_publish_views

```diff
@@ -2,11 +2,32 @@
 
 set -e
 
-# Fill in any missing view definitions in the moz-fx-data-shared-prod/ folder
-./script/generate_views 'moz-fx-data-shared-prod:*_stable.*'
+: "${TARGET_PROJECT:=moz-fx-data-shar-nonprod-efed}"
+: "${SQL_DIR:=sql/}"
 
-# All view definitions target moz-fx-data-shared-prod by default
-./script/publish_views sql/
+while [[ $# -gt 0 ]]
+do
+key="$1"
 
-# We additionally make sure we have identical view definitions in moz-fx-data-derived-datasets
-./script/publish_views --target-project moz-fx-data-derived-datasets sql/
+case $key in
+    --target-project)
+    TARGET_PROJECT="$2"
+    shift # past argument
+    shift # past value
+    ;;
+    --sql-dir)
+    SQL_DIR="$2"
+    shift # past argument
+    shift # past value
+    ;;
+esac
+done
+
+./script/generate_all_views --target-project "${TARGET_PROJECT}" --sql-dir "${SQL_DIR}"
+./script/publish_views --target-project "${TARGET_PROJECT}" "${SQL_DIR}"
+
+# We additionally make sure we have identical view definitions in moz-fx-data-derived-datasets and mozdata
+if [[ "$TARGET_PROJECT" == "moz-fx-data-shared-prod" ]]; then
+    ./script/publish_views --target-project moz-fx-data-derived-datasets "${SQL_DIR}"
+    ./script/publish_views --target-project mozdata "${SQL_DIR}"
```

Comment:
When publishing to mozdata here should the `--user-facing-only` flag be added to ensure only user-facing views are published?

```suggestion
    ./script/publish_views --user-facing-only --target-project mozdata "${SQL_DIR}"
```

Reply:
Any time we hard-code projects in bigquery-etl we lose the flexibility to support the [nonprod](https://github.com/mozilla-services/cloudops-infra/blob/master/projects/data-pioneer/Jenkinsfile.bigquery#L37) environment.

https://github.com/mozilla/bigquery-etl/issues/1110

Reply:
Indeed. This `generate_and_publish_views` script is intended to demonstrate usage. My intention is that Jenkins would simply move its existing call to `generate_views` with `./script/generate_all_views --target-project ${project} --sql-dir /tmp/sql`.

This generate_and_publish_views logic is mimicing what Jenkins does. Only for shared-prod does it do the extra publishing to mozdata and derived-datasets. This block is skipped for nonprod where we don't want to republish.